src/test/event.processor.spec.ts

```diff
@@ -0,0 +1,169 @@
+    await expect(eventProcessor.start({
+      emitterAddresses: ['erd1nz88q5pevl6up2qxsgpqgc0qmnm93lh888wwwa68kmz363kdwz9q8tnems'],
+      getLastProcessedTimestamp: async () => 37,
+      elasticUrl: 'https://index.multiversx.com',
+      onEventsReceived: async () => {},
+    })).rejects.toThrow(/Missing setLastProcessedTimestamp callback function/);
+  });
+
+  it('should work but no item matches the filters', async () => {
+    const eventProcessor = new EventProcessor();
+    let receivedEvents: any[] = [];
+    const mockResponse = createElasticSearchResponse();
+
+    // Mock axios.post to return a resolved Promise with mockResponse
+    (axios.post as jest.Mock).mockResolvedValue(mockResponse);
+
+    await eventProcessor.start({
+      emitterAddresses: ['erd1nz88q5pevl6up2qxsgpqgc0qmnm93lh888wwwa68kmz363kdwz9q8tnems'],
+      getLastProcessedTimestamp: async () => 37,
+      elasticUrl: 'https://myelastic.com',
+      onEventsReceived: async (highestTimestamp, events) => {receivedEvents = events;},
+      setLastProcessedTimestamp: async (timestamp: number) => {},
+    });
+
+    expect(receivedEvents?.length).toBe(0);
+    expect(axios.post).toHaveBeenCalledTimes(1);
+  });
+
+  it('should return no items if invalid Elasticsearch response', async () => {
+    let receivedEvents: any[] = [];
+
+    // Mock axios.post to return a resolved Promise with mockResponse
+    (axios.post as jest.Mock).mockResolvedValue({ invalid: 'response' });
+
+    await eventProcessor.start({
+      emitterAddresses: ['erd1nz88q5pevl6up2qxsgpqgc0qmnm93lh888wwwa68kmz363kdwz9q8tnems'],
+      getLastProcessedTimestamp: async () => 37,
+      elasticUrl: 'https://httpbin.com/404',
+      onEventsReceived: async (highestTimestamp, events) => {receivedEvents = events;},
+      setLastProcessedTimestamp: async (timestamp: number) => {},
+    });
+
+    expect(receivedEvents?.length).toBe(0);
+    expect(axios.post).toHaveBeenCalledTimes(1);
+  });
+
+  it('should work and receive events', async () => {
+    let receivedEvents: any[] = [];
+    const mockResponse1 = createElasticSearchResponse(123);
+    const mockResponse2 = createElasticSearchResponse();
+
+    // Mock axios.post to return a resolved Promise with mockResponse
+    (axios.post as jest.Mock)
+      .mockResolvedValueOnce(mockResponse1)
+      .mockResolvedValueOnce(mockResponse2);
+
+    await eventProcessor.start({
+      emitterAddresses: ['erd1nz88q5pevl6up2qxsgpqgc0qmnm93lh888wwwa68kmz363kdwz9q8tnems'],
+      getLastProcessedTimestamp: async () => 37,
+      elasticUrl: 'https://myelastic.com',
+      onEventsReceived: async (highestTimestamp, events) => {receivedEvents = events;},
+      setLastProcessedTimestamp: async (timestamp: number) => {},
+    });
+
+    expect(receivedEvents?.length).toBe(1);
+    expect(receivedEvents[0].timestamp).toEqual(123);
+    expect(axios.post).toHaveBeenCalledTimes(2);
+  });
+
```

Comment:
It would be useful to add a test that checks the behavior of the class when the call to `Elasticsearch` fails (eg using jest.MockRejectedValue to simulate a network error). This would cover cases of failure handling.

Reply:
Excelent! code coverage went to 98% :D